client/main.lua

```diff
@@ -21,31 +21,56 @@ end)
 -- TODO: Better implementation of this?
 RegisterCommand('vol', function(_, args)
 	if args[1] then
-		setVolume(args[1])
+		setVolume(args[1], args[2])
 	end
 end)
 
+RegisterCommand('volcheck', function(_, args)
+	TriggerEvent('chat:addMessage', {
+		color = { 255, 0, 0},
+		multiline = true,
+		args = {"Voice: ", "Proximity volume: "..math.floor(volumel * 100)..'%'}
+	  })
+	  TriggerEvent('chat:addMessage', {
+		color = { 255, 0, 0},
+		multiline = true,
+		args = {"Voice: ", "Call volume: "..math.floor(volumes.call * 100)..'%'}
+	  })
+	  TriggerEvent('chat:addMessage', {
+		color = { 255, 0, 0},
+		multiline = true,
+		args = {"Voice: ", "Radio volume: "..math.floor(volumes.radio * 100)..'%'}
+	  })
+end)
+
 --- function setVolume
 --- Toggles the players volume
 ---@param volume number between 0 and 100
 ---@param volumeType string the volume type (currently radio & call) to set the volume of (opt)
-function setVolume(volume, volumeType)
-	local volume = tonumber(volume)
-	local checkType = type(volume)
+function setVolume(vol, volumeType)
+	vol = tonumber(vol)
+	local checkType = vol and type(vol) or 'not number'
 	if checkType ~= 'number' then
 		return error(('setVolume expected type number, got %s'):format(checkType))
 	end
-	if volumeType then
+	vol = vol / 100
+	if volumeType and volumeType ~= 'proximity' then
 		local volumeTbl = volumes[volumeType]
 		if volumeTbl then
+			if volumeType == 'radio' then
+				volumes.radio = vol
+				SetResourceKvp("pmaVoiceRadio", vol)
+			elseif volumeType == 'call' then
+				volumes.call = vol
+				SetResourceKvp("pmaVoiceCall", vol)
+			end
 
 		else
 			error(('setVolume got a invalid volume type %s'):format(volumeType))
 		end
 	else
-		for types, vol in pairs(volumes) do
-			vol = volume
-		end
+		volumel = vol
+		SetResourceKvp('pmaVoiceMain', volumel)
```

Comment:
kvps should follow the formatting of the other kvps (so in this case it should be called `pma-voice_radioVolume` and `pma-voice_callVolume`

There also shouldn't be a 'main' volume, if this is set it should update the two volume keys

Reply:
These kvp's should also be set as floats, not strings see https://docs.fivem.net/natives/?_0x9ADD2938